src/python/WMCore/Services/pycurl_manager.py

```diff
@@ -56,11 +57,45 @@
 from io import BytesIO
 import http.client
 from urllib.parse import urlencode
+import traceback
+
+try:
+    import cStringIO as StringIO
+except ImportError: # python3
+    import io
+except:
+    import StringIO
 
 from Utils.Utilities import encodeUnicodeToBytes, decodeBytesToUnicode
 from Utils.PortForward import portForward, PortForward
 
 
+def decompress(body, headers):
+    encoding = ""
+    for h, v in headers.items():
+        if h.lower() == 'content-encoding' and 'gzip' in v.lower():
+            encoding = 'gzip'
+            break
+    if encoding != 'gzip':
+        return body
+
+    try:
+        if  sys.version.startswith('3.'):
+            zbuf = io.BytesIO()
+        else:
+            zbuf = StringIO.StringIO()
+        zbuf.write(body)
+        zbuf.seek(0)
+        zfile = gzip.GzipFile(mode='rb', fileobj=zbuf)
+        data = zfile.read()
+        zfile.close()
+        return data
+    except:
+        logger = logging.getLogger()
+        logger.error("HTTP headers %s" % headers)
+        logger.error("unable to decompress gzip content:\n%s" % traceback.format_exc())
```

Comment:
Instead of using traceback, I sugges to use `logger.exception`. Maybe you want to propagate the original error message as well, so something like:
```
except Exception as exc:
    msg = "Error details: {}".format(str(exc)) etc etc etc
```